web/main.go

```diff
@@ -180,7 +218,7 @@ func (s *server) authUser(r *http.Request) (user *pb.User, errFunc httpErrorWrit
 		return nil, badRequest, errors.Wrap(err, "failed to decode cookie")
 	}
 
-	userResp, err := s.getUser(userID)
+	userResp, err := s.getUser(trace.NewContext(ctx, span.NewChild("retrieve_user")), userID)
```

Comment:
The caller method is not aware of the caller but the calling context. In your current case, you are leaking that userRepos will need to mark a span to authUser handler. authUser shouldn't know about the span(s) getUser require.

We will document this ownership model as soon as APIs are more stable.